Marlin/src/module/planner.cpp

```diff
@@ -2575,14 +2623,18 @@ bool Planner::buffer_line(const float &rx, const float &ry, const float &rz, con
   #if IS_KINEMATIC
     inverse_kinematics(raw);
 
-    float mm = millimeters;
-    if (mm == 0.0) {
-      const float dx = rx - current_position[X_AXIS],
-                  dy = ry - current_position[Y_AXIS],
-                  dz = rz - current_position[Z_AXIS];
+    const float delta_mm_cart[] = {
+      rx - current_position[X_AXIS],
+      ry - current_position[Y_AXIS],
+      rz - current_position[Z_AXIS]
+      #if ENABLED(JUNCTION_DEVIATION)
+        , e - current_position[E_AXIS]
+      #endif
```

Comment:
A minor warning. The `current_position` does not have to be updated on a per-segment basis, and there may be code that calls this multiple times without ever updating `current_position` until the full move has been done. So we should check every call to `buffer_segment` to rule out this possibility.

Reply:
I've been thinking about that. One solution would be to only allow motion commands to call `planner.buffer_line` and not `planner.buffer_segment`. The planner could then keep a copy of the current Cartesian position, which is updated at the end of each call to `buffer_line`.

I believe I've already accomplished this on kinematic configurations, but UBL still calls `buffer_segment` for Cartesian machines.

Reply:
@thinkyhead - I just amended the commit to add `planner_cart` for kinematic machines only. Since `current_position` was only being used by kinematic machines, and they were already only using `buffer_line`, I think having the `Planner` track the current cartesian position mitigates your concern and should allow for accurate calculation of the delta between the target and current cartesian positions.